src/hotspot/share/jfr/leakprofiler/checkpoint/objectSampleWriter.cpp

```diff
@@ -625,6 +621,15 @@ ObjectSampleWriter::~ObjectSampleWriter() {
   write_array_infos(_writer);
   write_field_infos(_writer);
   write_root_descriptors(_writer);
+
+  if (field_infos != NULL) {
+    field_infos->clear();
+    field_infos = NULL;
+  }
+  sample_infos = NULL;
+  ref_infos = NULL;
+  array_infos = NULL;
+  root_infos = NULL;
```

Comment:
Maybe add a comment that these are RA allocated and therefore don't need to be destructed; otherwise this looks like a leak.